src/main/java/io/jenkins/plugins/opentelemetry/job/JunitAction.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The Original Author or Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.jenkins.plugins.opentelemetry.job;
+
+import hudson.model.InvisibleAction;
+import io.opentelemetry.api.common.AttributeKey;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * @see AttributeKey
+ * @see io.opentelemetry.api.common.AttributeType
+ */
+public class JunitAction extends InvisibleAction {
+
+    private transient Map<AttributeKey<?>, Object> attributes;
```

Comment:
What's happening when the Jenkins controller restarts during a pipeline execution? The attributes are lost?

Reply:
Yes, at the moment I didn't think about this feature. In one way or another, jenkins controller restarts don't support ephemeral workers, so I guess we don't need this feature yet.

We could add it if needed in a follow up, what do you think?

Reply:
Not anymore `transient `